pkg/sortition/sortition.go

```diff
@@ -96,23 +96,61 @@ func checkOperatorStatus(chain Chain) error {
 	}
 
 	if isLocked {
-		logger.Infof("sortition pool state is locked, waiting with the update")
+		logger.Info("sortition pool state is locked, waiting with the update")
 		return nil
 	}
 
 	if !isOperatorInPool {
-		logger.Infof("joining the sortition pool")
+		logger.Info("joining the sortition pool")
 		err := chain.JoinSortitionPool()
 		if err != nil {
 			logger.Errorf("could not join the sortition pool: [%v]", err)
 		}
 	} else {
-		logger.Infof("updating operator status in the sortition pool")
+		logger.Info("updating operator status in the sortition pool")
 		err := chain.UpdateOperatorStatus()
 		if err != nil {
 			logger.Errorf("could not update the sortition pool: [%v]", err)
 		}
 	}
 
+	err = checkRewardsEligibility(chain)
```

Comment:
The placement of this check is wrong - it does not cover all the cases. If the operator *is* in the sortition pool and *is* up to date, the `checkOperatorStatus` will return before reaching this line.

I would suggest checking `chain.IsEligibleForRewards()` after `chain.IsOperatorInPool()` and if the operator is in pool and is not eligible for rewards and if the pool is not locked, check if the eligibility can be restored.

Reply:
Ok, I also made an attempt to group this logic together, because we checked ex. `isOperatorInPool` a couple of times. The preserved existing logic with rewards restoration is done in [8402344](https://github.com/keep-network/keep-core/pull/3085/commits/8402344233845899565d138127d05c83868f6fb2). Grouping logic is done in [ac697ce](https://github.com/keep-network/keep-core/pull/3085/commits/ac697cee0d6d06b1ebc7c1648e5004d5f21089fc) & [bae2321](https://github.com/keep-network/keep-core/pull/3085/commits/bae23216447c73898a4c5692fce6ebdd08ac77f2) depending whether or not we need all the logging that is present now.